dali/test/python/test_operator_numpy_reader.py

```diff
@@ -124,25 +140,56 @@ def test_batch_file_list():
         arr_np_all = np.stack(arr_np_list, axis=0)
 
         # save filenames
-        with open(os.path.join(test_data_root, "input.lst"), "w") as f:
-            f.writelines("\n".join([os.path.basename(x) for x in filenames]))
+        file_list_path = os.path.join(test_data_root, "input.lst")
+        with open(file_list_path, "w") as f:
+            f.writelines("\n".join(filenames))
 
-        # create pipe
-        for num_threads in [1, 2, 4, 8]:
-            pipe = NumpyReaderPipeline(path = test_data_root,
-                                       file_list = os.path.join(test_data_root, "input.lst"),
-                                       batch_size = batch_size,
-                                       num_threads = num_threads,
-                                       device_id = 0)
-            pipe.build()
+        for device in ["cpu", "gpu"]:
+            if platform.processor() != "x86_64" and device == "gpu":
+                # GDS is supported only on x86_64
+                continue
+            yield check_batch, "", batch_size, num_samples, device , arr_np_all, file_list_path
 
-            for batch in range(0, num_samples, batch_size):
-                pipe_out = pipe.run()
-                arr_rd = pipe_out[0].as_array()
-                arr_np = arr_np_all[batch:batch+batch_size, ...]
 
-                # compare
-                assert_array_equal(arr_rd, arr_np)
+def test_batch_files():
+    with tempfile.TemporaryDirectory(prefix = gds_data_root) as test_data_root:
+        # create files
+        num_samples = 20
+        batch_size = 4
+        filenames = []
+        arr_np_list = []
+        for index in range(0,num_samples):
+            filename = os.path.join(test_data_root, "test_{:02d}.npy".format(index))
+            filenames.append(filename)
+            create_numpy_file(filename, (5, 2, 8), np.float32, False)
+            arr_np_list.append(np.load(filename))
+        arr_np_all = np.stack(arr_np_list, axis=0)
+
+        for device in ["cpu", "gpu"]:
+            if platform.processor() != "x86_64" and device == "gpu":
+                # GDS is supported only on x86_64
+                continue
+            yield check_batch, None, batch_size, num_samples, device , arr_np_all, None, filenames
+
+def test_batch_files_cache_headers():
+    with tempfile.TemporaryDirectory(prefix = gds_data_root) as test_data_root:
+        # create files
+        num_samples = 20
+        batch_size = 4
+        filenames = []
+        arr_np_list = []
+        for index in range(0,num_samples):
+            filename = os.path.join(test_data_root, "test_{:02d}.npy".format(index))
+            filenames.append(filename)
+            create_numpy_file(filename, (5, 2, 8), np.float32, False)
+            arr_np_list.append(np.load(filename))
+        arr_np_all = np.stack(arr_np_list, axis=0)
+
+        for device in ["cpu", "gpu"]:
+            if platform.processor() != "x86_64" and device == "gpu":
```

Comment:
and here. Since it's repeader maybe you can create a function ```is_gds_supported(platform)``` or something like that

Reply:
Done